### PR TITLE
ShellSort.java Add shell sort algorithm implementation in Java

### LINKED ISSUE #134: [Algo] Shell Sort [Java]
https://en.wikipedia.org/wiki/Shellsort

### PR DESCRIPTION
**Pull request by JonathanGin52**
This adds the shell sort algorithm. The method starts by sorting pairs of elements far apart from each other, then progressively reducing the gap between elements to be compared. Starting with far apart elements, it can move some out-of-place elements into position faster than a simple nearest neighbor exchange.

Closes https://github.com/NITSkmOS/Algorithms/issues/134

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #134 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

**Comment by sangamcse** [comment]
@JonathanGin52, Please add to the README table also.
And use `git commit --amend` while committing the changes.

File: shell_sort/ShellSort.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                while (inner > h - 1 && array[inner - h] >= temp) {
+                    array[inner] = array[inner - h];
+                    inner -= h;
+                }
+                array[inner] = temp;
+            }
+            h = (h - 1) / 3;
+        }
+    }
+}
```
Comment (sangamcse) [inline shell_sort/ShellSort.java]: Include EOL also. @JonathanGin52

Comment (JonathanGin52) [reply shell_sort/ShellSort.java]: Sorry, just to clarify, are you asking for a blank line at the end of the file?

Comment (sangamcse) [reply shell_sort/ShellSort.java]: Yes

Comment (JonathanGin52) [reply shell_sort/ShellSort.java]: could you take a look again?

**Comment by sangamcse** [comment]
https://github.com/NITSkmOS/Algorithms/pull/141#issuecomment-426064139

**Pull request closed without merging**